apps/lvgl_scanner/src/main.c

```diff
@@ -0,0 +1,333 @@
+        lv_label_ins_text(label, LV_LABEL_POS_LAST, "/2M");
+        break;
+    case BLE_HCI_LE_PHY_CODED:
+        lv_label_ins_text(label, LV_LABEL_POS_LAST, "/Coded");
+        break;
+    default:
+        break;
+    }
+
+    /* Set the name label if available and if it hasn't already been done */
+    if (fields.name != NULL && lv_obj_get_child(entry, 3) == NULL) {
+        label = lv_label_create(entry);
+        lv_obj_align(label, LV_ALIGN_BOTTOM_LEFT, 0, 0);
+        memcpy(name_buf, fields.name, fields.name_len);
+        name_buf[fields.name_len] = '\0';
+        lv_label_set_text_fmt(label, "%s", name_buf);
+    }
+}
+
+static void
+add_new_device(struct ble_gap_ext_disc_desc *desc)
+{
+    uint8_t id;
+
+    lv_obj_t *entry = lv_obj_create(entries_list);
+    lv_obj_set_size(entry, LV_PCT(100), 40);
+    lv_obj_add_style(entry, &style, 0);
+
+    id = lv_obj_get_child_cnt(entries_list) - 1;
+    devices[id].list_entry = entry;
+    devices[id].last_rssi = desc->rssi;
+    memcpy(devices[id].addr.val, desc->addr.val, 6);
+    fill_device_labels(desc, id, 1);
+}
+
+static void
+update_device(struct ble_gap_ext_disc_desc *desc, uint8_t id)
+{
+    uint8_t rssi_change;
+
+    fill_device_labels(desc, id, 0);
+
+    /* Sort the list if and update RSSI if it changed significantly */
+    rssi_change = abs(devices[id].last_rssi - desc->rssi);
+    if (rssi_change > 5) {
+        devices[id].last_rssi = desc->rssi;
+        sort_devices();
+    }
+}
+
+static uint8_t
+is_event_connectable(struct ble_gap_ext_disc_desc *desc)
+{
+    return desc->props & BLE_HCI_ADV_CONN_MASK;
+}
+
+static void
+list_init(void)
+{
+    entries_list = lv_obj_create(lv_scr_act());
+    lv_obj_set_size(entries_list, LV_PCT(100), LV_PCT(80));
+    lv_obj_align(entries_list, LV_ALIGN_TOP_MID, 0, 5);
+    lv_obj_set_flex_flow(entries_list, LV_FLEX_FLOW_COLUMN);
+}
+
+static void
+handle_disc_report(struct ble_gap_ext_disc_desc *desc)
+{
+    int id;
+
+    if (desc->rssi < -70 || !is_event_connectable(desc)) {
+        return;
+    }
+    if (lv_obj_get_child_cnt(entries_list) >= MAX_DEVICES) {
+        lv_obj_del(entries_list);
+        list_init();
+    }
+
+    id = get_device_id_by_addr(desc->addr.val);
+
+    if (id == DEVICE_ID_NONE) {
+        add_new_device(desc);
+        sort_devices();
+    } else {
+        update_device(desc, id);
+    }
+}
+
+static int
+scan_event(struct ble_gap_event *event, void *arg)
+{
+    switch (event->type) {
+    /* advertising report has been received during discovery procedure */
+    case BLE_GAP_EVENT_EXT_DISC:
+        handle_disc_report(&event->ext_disc);
+        return 0;
+
+    default:
+        MODLOG_DFLT(ERROR, "Discovery event not handled\n");
+        return 0;
+    }
+}
+
+static void
+start_scan(void)
+{
+    struct ble_gap_ext_disc_params coded;
+    struct ble_gap_ext_disc_params uncoded;
+    uint8_t filter_duplicates = 0;
+    uint8_t filter_policy = 0;
+    uint8_t limited = 0;
```

Comment:
not need to define variables which are in this case only used as parameter to function